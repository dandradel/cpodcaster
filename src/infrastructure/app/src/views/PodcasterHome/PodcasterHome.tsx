import React, { ReactElement } from 'react'
import Searcher from '../../components/Searcher/Searcher'
import Podcasts from '../../components/Podcasts/Podcasts'
import useError from '../../shared/hooks/useError'
import Error from '../../components/Error/Error'
import usePodcasts from '../../shared/hooks/usePodcasts'
import Skeleton from '../../components/Skeleton/Skeleton'

function PodcasterHome(): ReactElement {
  const podcasts = usePodcasts()
  const { error } = useError()
  const isPodcastsEmpty: boolean = podcasts?.length === 0

  if (error) {
    return <Error />
  }

  if (!podcasts) {
    return <Skeleton />
  }

  return (
    <>
      <Searcher chipLabel={podcasts.length} />
      {isPodcastsEmpty && <Error message='No podcasts found, try another search' />}
      <Podcasts podcasts={podcasts} />
    </>
  )
}

export default PodcasterHome
