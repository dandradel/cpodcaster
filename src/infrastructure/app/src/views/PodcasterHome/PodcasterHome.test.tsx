import React from 'react'
import '@testing-library/jest-dom'
import { cleanup, renderWithProviders, screen } from '../../shared/test/test-utils'
import PodcasterHome from './PodcasterHome'
import errorMock from '../../shared/__mocks__/error.json'

describe('PodcasterHome', () => {
  afterEach(cleanup)
  test('renders PodcasterHome', () => {
    const container = renderWithProviders(<PodcasterHome />)
    expect(container).toMatchSnapshot()
  })
  test('renders PodcasterHome with Searcher', () => {
    renderWithProviders(<PodcasterHome />)
    const searcher = screen.getByTestId('searcher-wrapper-testid')
    expect(searcher).toBeInTheDocument()
  })
  test('renders PodcasterHome with Podcasts', () => {
    renderWithProviders(<PodcasterHome />)
    const podcasts = screen.getByTestId('podcasts-wrapper-testid')
    expect(podcasts).toBeInTheDocument()
  })
  test('renders PodcasterHome with Error', () => {
    renderWithProviders(<PodcasterHome />, {
      preloadedState: {
        error: errorMock,
        podcasts: { podcasts: null, podcastCard: null },
        episodes: { episodes: null },
      },
    })
    const error = screen.getByText(
      'Oops something went wrong, try reloading the page again please :)',
    )
    expect(error).toBeInTheDocument()
  })
})
