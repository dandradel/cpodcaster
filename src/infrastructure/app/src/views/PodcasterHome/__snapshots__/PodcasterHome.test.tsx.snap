// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PodcasterHome renders PodcasterHome 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiBox-root css-zyoky0"
        data-testid="searcher-wrapper-testid"
      >
        <div
          class="MuiFormControl-root MuiTextField-root css-x66ii1-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined css-1pysi21-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
            for=":r0:"
            id=":r0:-label"
          >
            Filter podcasts...
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
              id=":r0:"
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-yjsfm1"
              >
                <span>
                  Filter podcasts...
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          class="MuiChip-root MuiChip-filled MuiChip-sizeSmall MuiChip-colorPrimary MuiChip-filledPrimary css-vkz45j-MuiChip-root"
          data-testid="search-chip-number-testid"
        >
          <span
            class="MuiChip-label MuiChip-labelSmall css-wjsjww-MuiChip-label"
          >
            1
          </span>
        </div>
      </div>
      <div
        class="MuiBox-root css-170i8fq"
        data-testid="podcasts-wrapper-testid"
      >
        <div
          class="MuiBox-root css-cl0zsv"
          data-testid="podcastCover-wrapper-testid"
        >
          <div
            class="MuiBox-root css-0"
            data-testid="podcastCover-container-testid"
          >
            <img
              class="podcast__image"
              data-testid="podcast-image-testid"
              src="https://picsum.photos/200/300"
            />
            <div
              class="MuiBox-root css-1av9gx5"
            >
              <h6
                class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom css-18k87ye-MuiTypography-root"
                data-testid="podcast-name-testid"
              >
                Podcast 1
              </h6>
              <h6
                class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-gutterBottom css-7ft05y-MuiTypography-root"
                data-testid="podcast-author-testid"
              >
                Author 1
              </h6>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiBox-root css-zyoky0"
      data-testid="searcher-wrapper-testid"
    >
      <div
        class="MuiFormControl-root MuiTextField-root css-x66ii1-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined css-1pysi21-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          for=":r0:"
          id=":r0:-label"
        >
          Filter podcasts...
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
            id=":r0:"
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-yjsfm1"
            >
              <span>
                Filter podcasts...
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        class="MuiChip-root MuiChip-filled MuiChip-sizeSmall MuiChip-colorPrimary MuiChip-filledPrimary css-vkz45j-MuiChip-root"
        data-testid="search-chip-number-testid"
      >
        <span
          class="MuiChip-label MuiChip-labelSmall css-wjsjww-MuiChip-label"
        >
          1
        </span>
      </div>
    </div>
    <div
      class="MuiBox-root css-170i8fq"
      data-testid="podcasts-wrapper-testid"
    >
      <div
        class="MuiBox-root css-cl0zsv"
        data-testid="podcastCover-wrapper-testid"
      >
        <div
          class="MuiBox-root css-0"
          data-testid="podcastCover-container-testid"
        >
          <img
            class="podcast__image"
            data-testid="podcast-image-testid"
            src="https://picsum.photos/200/300"
          />
          <div
            class="MuiBox-root css-1av9gx5"
          >
            <h6
              class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom css-18k87ye-MuiTypography-root"
              data-testid="podcast-name-testid"
            >
              Podcast 1
            </h6>
            <h6
              class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-gutterBottom css-7ft05y-MuiTypography-root"
              data-testid="podcast-author-testid"
            >
              Author 1
            </h6>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
